introspection-engine/connectors/mongodb-introspection-connector/src/lib.rs

```diff
@@ -59,6 +61,37 @@ impl MongoDbIntrospectionConnector {
     fn database(&self) -> Database {
         self.connection.database(&self.database)
     }
+
+    async fn describe(&self, preview_features: BitFlags<PreviewFeature>) -> ConnectorResult<MongoSchema> {
+        let mut schema = mongodb_schema_describer::describe(&self.connection, &self.database)
+            .await
+            .map_err(crate::Error::from)?;
+
+        if !preview_features.contains(PreviewFeature::FullTextIndex) {
+            #[allow(clippy::needless_collect)] // well, mr. clippy, maybe you should read about the borrow checker...
+            let kept_indexes: Vec<_> = schema.drain_indexes().filter(|i| !i.is_fulltext()).collect();
```

Comment:
Doesn't that invalidate index ids?

Reply:
No, we only have them in one collection, and then keep the indexes in a vector. This empties both:

https://github.com/prisma/prisma-engines/pull/2449/files#diff-cf78c2b811a9b0c673afa5262b4852b1dc00ab5d1b4ca84fc31c5599d427c37eR132-R135